DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -50,42 +62,73 @@ public IActionResult EditDetails()
             var userDelegateId = User.GetNullableCandidateId();
             var (adminUser, delegateUser) = userService.GetUsersById(userAdminId, userDelegateId);
 
-            var model = new EditDetailsViewModel(adminUser, delegateUser);
+            var jobGroups = jobGroupsDataService.GetJobGroupsAlphabetical().ToList();
+            ViewBag.JobGroupOptions =
+                SelectListHelper.MapOptionsToSelectListItemsWithSelectedText(jobGroups, delegateUser?.JobGroupName);
+            ViewBag.CustomFields = GetCustomFieldsWithDelegateAnswers(delegateUser);
+
+            var model = new EditDetailsViewModel(adminUser, delegateUser, jobGroups);
 
             return View(model);
         }
 
         [HttpPost]
         public IActionResult EditDetails(EditDetailsViewModel model, string action)
         {
+            ViewBag.JobGroupOptions = GetJobGroupItems(model.JobGroupId);
+            ViewBag.CustomFields = GetCustomFieldsWithEnteredAnswers(model);
             return action switch
             {
                 "save" => EditDetailsPostSave(model),
                 "previewImage" => EditDetailsPostPreviewImage(model),
                 "removeImage" => EditDetailsPostRemoveImage(model),
-                _ => View(model),
+                _ => View(model)
             };
         }
 
         private IActionResult EditDetailsPostSave(EditDetailsViewModel model)
         {
+            ValidateCustomPrompts(model);
+
             if (!ModelState.IsValid)
             {
                 return View(model);
             }
 
             if (model.ProfileImageFile != null)
             {
-                ModelState.AddModelError(nameof(EditDetailsViewModel.ProfileImageFile), "Preview your new profile picture before saving");
+                ModelState.AddModelError(nameof(EditDetailsViewModel.ProfileImageFile),
+                    "Preview your new profile picture before saving");
                 return View(model);
             }
 
             var userAdminId = User.GetAdminId();
             var userDelegateId = User.GetNullableCandidateId();
 
-            if (!userService.TryUpdateUserAccountDetails(userAdminId, userDelegateId, model.Password,  model.FirstName, model.LastName, model.Email, model.ProfileImage))
+            var accountDetailsData = new AccountDetailsData(userAdminId,
+                userDelegateId,
+                model.Password!,
+                model.FirstName!,
+                model.LastName!,
+                model.Email!,
+                model.ProfileImage);
+
+            var centreAnswersData = userDelegateId == null
+                ? null
+                : new CentreAnswersData(
+                    User.GetCentreId(),
+                    model.JobGroupId!.Value,
+                    model.Answer1,
+                    model.Answer2,
+                    model.Answer3,
+                    model.Answer4,
+                    model.Answer5,
+                    model.Answer6);
+

```

Comment:
Might be worth pulling this into something like:
`private (AccountDetailsData, CentreAnswersData?) MapToUpdateAccountData(EditDetailsViewModel model, int? userAdminId, int? userDelegateId)`